phpBB/phpbb/console/command/searchindex/create.php

```diff
@@ -0,0 +1,163 @@
+use phpbb\search\search_backend_factory;
+use phpbb\search\state_helper;
+use phpbb\user;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class create extends command
+{
+	/** @var config */
+	protected $config;
+
+	/** @var language */
+	protected $language;
+
+	/** @var log */
+	protected $log;
+
+	/** @var post_helper */
+	protected $post_helper;
+
+	/** @var search_backend_factory */
+	protected $search_backend_factory;
+
+	/** @var state_helper */
+	protected $state_helper;
+
+	/**
+	 * Construct method
+	 *
+	 * @param config					$config
+	 * @param language					$language
+	 * @param log						$log
+	 * @param post_helper				$post_helper
+	 * @param search_backend_factory	$search_backend_factory
+	 * @param state_helper				$state_helper
+	 * @param user						$user
+	 */
+	public function __construct(config $config, language $language, log $log, post_helper $post_helper, search_backend_factory $search_backend_factory, state_helper $state_helper, user $user)
+	{
+		$this->config = $config;
+		$this->language = $language;
+		$this->log = $log;
+		$this->post_helper = $post_helper;
+		$this->search_backend_factory = $search_backend_factory;
+		$this->state_helper = $state_helper;
+
+		parent::__construct($user);
+	}
+
+	/**
+	 * Sets the command name and description
+	 *
+	 * @return void
+	 */
+	protected function configure()
+	{
+		$this
+			->setName('searchindex:create')
+			->setDescription($this->language->lang('CLI_DESCRIPTION_SEARCHINDEX_CREATE'))
+			->addArgument(
+				'search-backend',
+				InputArgument::REQUIRED,
+				$this->language->lang('CLI_SEARCHINDEX_SEARCH_BACKEND_NAME')
+			)
+		;
+	}
+
+	/**
+	 * Executes the command searchindex:create
+	 *
+	 * Create search index
+	 *
+	 * @param InputInterface	$input  The input stream used to get the options
+	 * @param OutputInterface	$output The output stream, used to print messages
+	 *
+	 * @return int 0 if all is well, 1 if any errors occurred
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output): int
+	{
+		$io = new SymfonyStyle($input, $output);
+
+		$io->section($this->language->lang('CLI_DESCRIPTION_SEARCHINDEX_CREATE'));
+
+		$search_backend = $input->getArgument('search-backend');
+
+		try
+		{
+			$search = $this->search_backend_factory->get($search_backend);
+			$name = $search->get_name();
+		}
+		catch (no_search_backend_found_exception $e)
+		{
+			$io->error($this->language->lang('CLI_SEARCHINDEX_BACKEND_NOT_FOUND', $search_backend));
+			return command::FAILURE;
+		}
+
+		if ($this->state_helper->is_action_in_progress())
+		{
+			$io->error($this->language->lang('CLI_SEARCHINDEX_ACTION_IN_PROGRESS', $search_backend));
+			return command::FAILURE;
+		}
+
+		try
+		{
+			$progress = $this->create_progress_bar($this->post_helper->get_max_post_id(), $io, $output, true);
```

Comment:
```suggestion
			$progress = $this->create_progress_bar($count_of_all_post_ids, $io, $output, true);
```
Should not the the `$max` parameter of `create_progress_bar(...)` ,
be total no of posts (SQL: `COUNT(post_id)`) ,
instead of max post id (SQL: `MAX(post_id)`) ?

Reply:
No, because post_counter reference the last indexed post id, not the number of posts indexed

Reply:
I mean something like this:
https://github.com/phpbb/phpbb/pull/6328/commits/b4a65a8f596f901377d9297590b18d2ed2f20050#diff-2f31352ab73095253f68c8cd43b1065cd218084bbadea4deacd66deec5d3e0f2R601-R616

Reply:
Yes, that's another way of doing it
I don't have a preference, I just kept it like that because that was the data I had without having to add extra querys.
Are those querys efficent with a large quantity of posts?

Reply:
Yes! you are right,
to have 2 SQL `COUNT(...)` quires for each batch will be inefficient for CLI.

Reply:
The same for web i think

Reply:
No! I am displaying the no of post pending from total.
EG. Lang Str:
https://github.com/phpbb/phpbb/commit/89fac5268632323edda47dc5876f3e1066399a93#diff-3646dfa09c7de93c26c98ce00fbe07be7c297e219e34055839527e049f47bc23R128